File: Program16.c
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//	Write a program which accepts N numbers from user and accept one another number as NO and returns the
//	index of last occurence of that NO.

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<stdio.h>
#include<stdlib.h>
#include<unistd.h>

typedef signed int s_bool;

#define TRUE 1
#define FALSE 0

int searchLastOccurence(int iLength,int iArr[],int iNum)
{
	register int iCnt = 0;
	auto int iIndex = -1;
	auto s_bool bFlag = FALSE;
	
	if(iNum > 100)
	{
		_exit(EXIT_FAILURE);
	}
	
	for(iCnt = 0; iCnt < iLength; iCnt++)
	{

		if(iArr[iCnt] == iNum)
		{
			iIndex = iCnt;
		}

	}
	return iIndex+1;
}

int main(void)
{
	int searchLastOccurence(int,int *,int);

	auto int iSize = 0,iNo = 0,iCnt = 0,iRet = 0;
	auto int *iPtr = NULL;
	
	printf("Enter number of elements you want to store in array : \n");
	scanf("%d",&iSize);
	
	iPtr = (int *)malloc(iSize * sizeof(int));
	if(iPtr == NULL)
	{
		perror("malloc");
		_exit(EXIT_FAILURE);
	}
	
	printf("\nEnter elements : \n");
	for(iCnt = 0; iCnt < iSize; iCnt++)
	{
		scanf("%d",&iPtr[iCnt]);
	}
	
	printf("\nElements stored in array are : \n");
	for(iCnt = 0; iCnt < iSize; iCnt++)
	{
		printf("%d\t",iPtr[iCnt]);
	}
	printf("\n");
	
	printf("\nEnter number that you want to search last occurence : \n");
	scanf("%d",&iNo);
	
	iRet = searchLastOccurence(iSize,iPtr,iNo);
	if(iRet == -1)
	{
		printf("\n%d is not present in the array\n",iNo);
	}
	else
	{	
		printf("\n%d last occured at %d position\n",iNo,iRet);
	}
	
	if(iPtr)
	{
		free(iPtr);
	}
	
	exit(EXIT_SUCCESS);

}

File: Program16.java
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//	Write a program which accepts N numbers from user and accept one another number as NO and returns the
//	index of last occurence of that NO.

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class ArrayX
{

	private int iArr[];
	
	public ArrayX(int iSize)
	{
		iArr = new int[iSize];
	}
	
	public void acceptDisplay()
	{
		Scanner sObj = new Scanner(System.in);
		
		int iCnt = 0;
		
		System.out.println("\nEnter elements : ");
		for(iCnt = 0; iCnt < iArr.length; iCnt++)
		{
			iArr[iCnt] = sObj.nextInt();
		}
		
		System.out.println("\nElements stored in array are : ");
		for(iCnt = 0; iCnt < iArr.length; iCnt++)
		{
			System.out.print(iArr[iCnt]+"\t");
		}
		System.out.println();
		
	}
	
	public int searchLastOccurence(int iNum)
	{
		int iCnt = 0,iIndex = -1;
		
		for(iCnt = 0; iCnt < iArr.length; iCnt++)
		{
			if(iNum == iArr[iCnt])
			{
				iIndex = iCnt;
			}
		}
		return iIndex+1;
	}
	
}

class Program16
{

	public static void main(java.lang.String Arg[])
	{
		int iValue = 0,iNo = 0,iRet = 0;
		
		Scanner sObj = new Scanner(System.in);
		
		System.out.println("Enter number of elements you want to store in array : ");
		iValue = sObj.nextInt();
		
		ArrayX aObj = new ArrayX(iValue);
		
		aObj.acceptDisplay();		
		
		System.out.println("\nEnter number that you want to search last occurence : ");
		iNo = sObj.nextInt();
		
		iRet = aObj.searchLastOccurence(iNo);
		System.out.println("\n"+iNo+" last occured at "+iRet+" position\n");			
			
		sObj.close();
	}
}
